### ISSUE TITLE
hover doesn't seem to work

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran-lsp#24]
It doesn't show a definition:

![image](https://github.com/user-attachments/assets/4cac657f-f034-4c39-9251-838e30de14f0)

**Comment by dylon** [comment]
For this text, lfortran returns that there are two symbols: `expr2` and `program`. `expr2` begins on line 0, character 1 and ends on line 8, character 11. `program` begins on line 3, character 12 and ends on line 3, character 18. If we index the symbols by identifier, when the cursor hovers over `program`, we look up that its definition consists of only the 7 characters that form its name. That is why the hover preview is not working correctly. I will file an lfortran bug accordingly.

```shell
$ ./src/bin/lfortran --show-document-symbols tests/expr2.f90 | jq
[
  {
    "kind": 12,
    "location": {
      "range": {
        "start": {
          "character": 1,
          "line": 0
        },
        "end": {
          "character": 11,
          "line": 8
        }
      },
      "uri": "uri"
    },
    "name": "expr2"
  },
  {
    "kind": 13,
    "location": {
      "range": {
        "start": {
          "character": 12,
          "line": 3
        },
        "end": {
          "character": 18,
          "line": 3
        }
      },
      "uri": "uri"
    },
    "name": "program"
  }
]
```

**Comment by certik** [comment]
I can see a couple issues here:
* numbering lines from 0, they should be numbered from 1
* returning `expr` as the declaration, which it should not be at all I think

CC @Pranavchiku

**Comment by Pranavchiku** [comment]
Checking this

**Comment by certik** [comment]
The above example now shows:

<img width="315" alt="image" src="https://github.com/user-attachments/assets/c0d82dc4-42fd-4bcd-8094-45859558fec8" />

Which is the wrong symbol --- it is showing the main program symbol, not the actual local variable `program` (of the same name). So this is incorrect.

And if you rename the variable to have a unique name, it won't show anything.

**Comment by Pranavchiku** [comment]
Are the following two expected behavior? If yes, https://github.com/lfortran/lfortran-lsp/pull/37 closes it.

<img width="455" alt="image" src="https://github.com/user-attachments/assets/4305c466-d083-4547-8a6b-f96f12a8dde9" />
<img width="478" alt="image" src="https://github.com/user-attachments/assets/69bd6edb-0b64-4727-b6ad-9425a7ee05a4" />

**Comment by Pranavchiku** [comment]
Closing with reference to #37

**Comment by certik** [comment]
Hovering over a variable still doesn't seem to work (https://github.com/lfortran/lfortran-lsp/issues/49), but hovering over a function call now works:

<img width="271" alt="image" src="https://github.com/user-attachments/assets/30aa1124-1b2a-4525-b5a8-e75449190697" />

The above program for `program` also doesn't work, but that's due to #49, so let's leave this issue closed, and do improvements of this in #49.
